// @hapiness/core/core/decorators.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const decorators_1 = require("../externals/injection-js/util/decorators");
exports.makeDecorator = decorators_1.makeDecorator;
const type_1 = require("../externals/injection-js/facade/type");
exports.Type = type_1.Type;
const injection_js_1 = require("../externals/injection-js");
exports.Inject = injection_js_1.Inject;
exports.Injectable = injection_js_1.Injectable;
exports.InjectionToken = injection_js_1.InjectionToken;
exports.Optional = injection_js_1.Optional;
/**
 * Create a decorator with metadata
 *
 * @param  {string} name
 * @param  {{[name:string]:any;}} props?
 * @returns CoreDecorator
 */
function createDecorator(name, props) {
    return decorators_1.makeDecorator(name, props);
}
exports.createDecorator = createDecorator;
exports.HapinessModule = createDecorator('HapinessModule', {
    version: undefined,
    declarations: undefined,
    providers: undefined,
    imports: undefined,
    exports: undefined
});
exports.Lib = decorators_1.makeDecorator('Lib', null);
